content-src/components/DiscoveryStreamBase/DiscoveryStreamBase.jsx

```diff
@@ -168,13 +169,63 @@ export class _DiscoveryStreamBase extends React.PureComponent {
   render() {
     // Select layout render data by adding spocs and position to recommendations
     const layoutRender = selectLayoutRender(this.props.DiscoveryStream, rickRollCache);
-    const styles = [];
     const {spocs, feeds} = this.props.DiscoveryStream;
-
     if (!spocs.loaded || !feeds.loaded) {
       return null;
     }
 
+    // Find the first component of a type and remove it from layout
+    const extractComponent = type => {
+      for (const [rowIndex, row] of Object.entries(layoutRender)) {
+        for (const [index, component] of Object.entries(row.components)) {
+          if (component.type === type) {
+            // Remove the row if it was the only component or the single item
+            if (row.components.length === 1) {
+              layoutRender.splice(rowIndex, 1);
+            } else {
+              row.components.splice(index, 1);
+            }
+            return component;
+          }
+        }
+      }
+      return null;
+    };
+
+    // Extract TopSites to render before the rest and Message to use for header
+    const topsites = extractComponent("TopSites");
+    const message = extractComponent("Message");
+
+    // Render a DS-style TopSites then the rest in a section that reuses the
+    // "topstories" Section state
+    const topStories = this.props.Sections.find(s => s.id === "topstories");
+    return (
+      <React.Fragment>
+        {topsites && this.renderLayout([{
+          width: 12,
+          components: [topsites],
+        }])}
+        <CollapsibleSection
```

Comment:
I think Tim mentioned this feature being something we could turn on and off via a pref, not sure if we still want to go that route though.

Reply:
Tim suggests that it'll be good to have something as part of `.config` pref to turn off if we need to. Probably a good idea as without `collapsible: true` some layouts wouldn't be possible.